client/flutter/lib/components/updated_app_bar.dart

```diff
@@ -0,0 +1,1361 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart' hide AppBar;
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+// COPIED FROM flutter/lib/src/material/app_bar.dart to add excludeHeaderSemantics
```

Comment:
Give an exact github url that you copied it from, with github commit sha.

Also do you need to update the THIRD_PARTY_LICENSE file?

Reply:
err, it's from the Flutter repo so a copy of the file without some minor changes is already in the app. Is it still necessary?

Reply:
Def need the exact GitHub url with commit.

Wrt to THIRD_PARTY_LICENSE I'd have to see if the *exact* same license is already in THIRD_PARTY_LICENSE.  If it is I usually just add the extra path.

Reply:
@britannio Are you sure the link is correct?

It starts with: `// Copyright 2014 The Flutter Authors. All rights reserved.` while yours has `// Copyright 2015 The Chromium Authors. All rights reserved.`